ReSharper.FSharp/src/FSharp.TypeProviders.Protocol/src/TypeProvidersExternalProcessFactory.cs

```diff
@@ -1,38 +1,40 @@
 ﻿using JetBrains.Annotations;
 using JetBrains.Application.Processes;
 using JetBrains.Application.Threading;
-using JetBrains.Diagnostics;
 using JetBrains.Lifetimes;
 using JetBrains.Platform.MsBuildHost;
 using JetBrains.ProjectModel;
-using JetBrains.ProjectModel.Build;
-using JetBrains.ProjectModel.Properties;
-using JetBrains.ReSharper.Resources.Shell;
 using JetBrains.Util;
 
 namespace JetBrains.ReSharper.Plugins.FSharp.TypeProviders.Protocol
 {
+  public interface IProjectsRequiringFrameworkVisitor
```

Comment:
The current name is somewhat misleading, as the component actually cares about F# projects only. We also probably don't want to say it's implemented as a visitor, as that is an implementation detail.

Maybe `FSharpProjectsRequiringFrameworkCache` or `FSharpProjectsRequiringFramework`?